thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -62,11 +62,37 @@ export default Service.extend({
       offsetsAndUrns.push([toAbsoluteUrn(urn, requestContext.compareMode).split(':')[2].toLowerCase(), urn]);
       metricUrnToOffestAndUrn[metricUrn] = offsetsAndUrns;
     });
-
-    Object.keys(metricUrnToOffestAndUrn).forEach(
+    const metricUrns = Object.keys(metricUrnToOffestAndUrn);
+    metricUrns.forEach(
       metricUrn => {
-        return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        return this._cacheRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
       });
+    metricUrns.forEach(
+        metricUrn => {
+          return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        });
+  },
+
+
+  /**
+   * Caches the metric data for a row of the metric table
+   *
+   * @param {String} metricUrn Metric urn
+   * @param {Object} context Context
+   * @param {Object} metricUrnToOffestAndUrn Hash map from metric urn to offset and urn
+   * @returns {undefined}
+   */
+  async _cacheRowSlice(metricUrn, context, metricUrnToOffestAndUrn) {
+    const [start, end] = context.anomalyRange;
```

Comment:
syntax: space before and after `[ a, b ]` when we destructuring arrays or objects `const { keyProp } = objToDestructure`

Reply:
Comment applies to whole pr

Reply:
do we not use prettier? All these syntax issues would have been all auto fix by prettier.

Reply:
let me setup to automate this. It will auto update with prettier for any file changed.

Reply:
Thanks! That would be much easier!